designer/client/section-create.js

```diff
@@ -1,62 +1,83 @@
 import React from 'react'
 import { clone } from 'digital-form-builder-model/lib/helpers'
+import { camelCase } from './helpers'
 
 class SectionCreate extends React.Component {
   state = {}
 
-  onSubmit = e => {
+  async onSubmit (e) {
     e.preventDefault()
-    const form = e.target
-    const formData = new window.FormData(form)
-    const name = formData.get('name').trim()
-    const title = formData.get('title').trim()
+    const { name, title, generatedName } = this.state
     const { data } = this.props
     const copy = clone(data)
 
-    const section = { name, title }
-    copy.sections.push(section)
+    const updated = copy.addSection(name || generatedName, title.trim())
 
-    data.save(copy)
-      .then(data => {
-        console.log(data)
-        this.props.onCreate({ data })
-      })
-      .catch(err => {
-        console.error(err)
-      })
+    try {
+      const savedData = await data.save(updated)
+      this.props.onCreate(savedData)
+    } catch (err) {
+      console.error(err)
+    }
   }
 
   onBlurName = e => {
     const input = e.target
-    const { data } = this.props
     const newName = input.value.trim()
+    const { data } = this.props
 
     // Validate it is unique
     if (data.sections.find(s => s.name === newName)) {
       input.setCustomValidity(`Name '${newName}' already exists`)
     } else {
       input.setCustomValidity('')
     }
+    this.setState({
+      name: newName
+    })
+  }
+
+  onChangeTitle = e => {
+    const input = e.target
+    const { data } = this.props
+    const newTitle = input.value
+    const generatedName = camelCase(newTitle).trim()
+    let newName = generatedName
+
+    let i = 1
+    while (data.sections.find(s => s.name === newName)) {
```

Comment:
you can use `for (let i = 0; data.sections.find(s => s.name === newName) ; i++) {`. wont block merge for this though

Reply:
cool, thanks for the tip!